### DISCUSSION TITLE
Dynamically load a new locale on runtime?

### DISCUSSION THREAD

**Discussion by alkstal** [ivanhofer/typesafe-i18n#663] [Q&A] [answered]
I know this is probably a bad idea considering how typesafe-i18n works, but I've been asked if it's possible and I would like to provide a good answer to it 😄 

Would it be possible to get a `Translation` from an API and load it on runtime without saving anything to disk?

Given I have the `Translation` object fetched and ready from an API, could I make it work on runtime?

I've seen that `loadLocale` only really works with the files that exist on disk and that the importer saves and runs the generator.
Everything generated only knows about what's on disk.

But would it be possible in some ugly, hacky way to add a translation and have everything work though?
Or is it not possible to do in a working way?

My own explorations say that no, this is not possible to do, but I might have missed something and in reality it could be easier than I thought. 😄

**Comment by ivanhofer** [comment, marked as answer]
By default `typesafe-i18n` generates a lot of boilerplate code for you. This works fine for 95% of projects I guess. If you want to customize the behavior, just set the [`generateOnlyTypes`-option](https://github.com/ivanhofer/typesafe-i18n/tree/main/packages/generator#generateonlytypes) to `true` and modify the files like you need them.

If you just want to load locales in a different way you can also manually set the `loadedLocales` object inside `i18n/i18n-util.ts`.

If you really want to opt opt of the typesafety features, you can load `Translations` via a ´fetch` calls in runtime, but you will not know if the translated strings contain an error.

I know of some companies that use a combined approach. They save translations to the Git repo once per day to check if translations contain an error, but also load the translations on runtime once per hour to get the latest strings. Maybe this is a solution you want to apply to your project too.

**Reply by alkstal** [reply]
Thanks for the reply, it was very helpful
